### ISSUE TITLE
Binaries for multiple linux architectures

### ISSUE THREAD

**Issue by andrjohns** [stan-dev/stanc3#926]
As part of the linux packaging stuff I'm working on, I've got a bunch of chroot environments building stanc3 binaries for multiple linux architectures (everything listed as 'Supported' by debian):
- `arm64`
- `armel`
- `armhf`
- `i386`
- `amd64`
- `mipsel`
- `mipsel64`
- `ppc64el`
- `s390x`

Is this something that would be useful to incorporate in the stanc3 release process? Otherwise I can just build and host them on my github, and we can point people there if they want binaries for other architectures

**Comment by rok-cesnovar** [comment]
Nice! We currently build only the arm64 one in addition to the x86 binary. But right now we use a dedicated arm64 instance for that, which is a bit annoying.

How difficult is this to set up?

**Comment by andrjohns** [comment]
It's surprisingly painless! Here's a basic example of setting up the chroot for `arm64`:
```
# Create folder and initialise chroot through qemu
mkdir -p /var/chroot/debian/arm64
debootstrap --arch arm64 bullseye /var/chroot/debian/arm64 http://deb.debian.org/debian/
mount -t proc proc /var/chroot/debian/arm64/proc

# Bind mount stanc3 folder to be accessed inside chroot
mkdir -p /var/chroot/debian/arm64/home/deb_build/stanc3
mount --bind ~/.stanc3/stanc3/ /var/chroot/debian/arm64/home/deb_build/stanc3

# Enter arm64 chroot
chroot /var/chroot/debian/arm64

# Install opam and dependencies
apt update
apt install opam curl bzip2 git tar curl ca-certificates openssl \\
            m4 bash libpcre3-dev -y

# Setup build environment
cd /home/deb_build/stanc3
opam init --disable-sandboxing -y
eval $(opam env)
opam switch create 4.07.0
opam switch 4.07.0
eval $(opam env)
bash -x scripts/install_build_deps.sh
eval $(opam env)
```

And it's ready to build!

There are some quirks though. `dune subst` and parts of the testing script don't seem to like being run under qemu chroot and error in strange ways. So it's setup to run `dune subst` outside of the chroot, so that it's processed and ready to be accessed inside the chroot. I haven't figured out a fix for the testing framework yet, but the bernoulli model is building and running without issue

**Comment by andrjohns** [comment]
I didn't think of this yesterday, but how about I look into putting together a docker image with the chroots all setup and ready to access? That would make integrating with the current build process much simpler

**Comment by andrjohns** [comment]
@serban-nicusor-toptal Would it be possible to enable `binfmt-support` on our Docker host? Then we could use multiarch docker images under qemu to build the binaries: https://hub.docker.com/r/multiarch/alpine

**Comment by serban-nicusor-toptal** [comment]
Of course @andrjohns I'll look into it later today

**Comment by serban-nicusor-toptal** [comment]
Hey @andrjohns it seems to be already enabled on our Docker hosts.

How did I check?
1. Make sure it's installed: `sudo apt-get install -y binfmt-support`
2. Check if it's mounted: `grep binfmt /proc/mounts`
3. Check if it's enabled: `cat /proc/sys/fs/binfmt_misc/status`

[source](https://access.redhat.com/solutions/1985633)

Please let me know if there's anything else I need to do to make this work.

**Comment by andrjohns** [comment]
Excellent thanks!

**Comment by andrjohns** [comment]
Closed by #936
